package NYT.NScheduler.NProto;

////////////////////////////////////////////////////////////////////////////////

message TJobResources
{
    optional int32 user_slots = 1;
    optional double cpu = 2;
    optional int64 memory = 3;
    optional int32 network = 4;
}

message TExecNodeDescriptor
{
    required int32 node_id = 1;
    required string address = 2;
    required double io_weight = 3;
    required TJobResources resource_limits = 4;
    repeated string tags = 5;
}

message TExecNodeDescriptorList
{
    repeated TExecNodeDescriptor exec_nodes = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqHeartbeat
{
}

message TRspHeartbeat
{
    optional TExecNodeDescriptorList exec_nodes = 1;
}

////////////////////////////////////////////////////////////////////////////////

